Source/SteveSloth/Private/MyProjectileBaseClass.cpp

```diff
@@ -15,32 +15,213 @@ AMyProjectileBaseClass::AMyProjectileBaseClass()
 	PrimaryActorTick.bCanEverTick = true;
 
 	ProjectileMovement = CreateDefaultSubobject<UProjectileMovementComponent>(TEXT("Projectile Movement Component"));
+	ProjectileHitbox = CreateDefaultSubobject<UBoxComponent>(TEXT("Hitbox"));
+	AreaOfEffectHitbox = CreateDefaultSubobject<USphereComponent>(TEXT("AOEHitbox"));
 
 	// Initialize Variables to the Projectile Data Table
 	const auto projectileData = ProjectileDataTable.GetRow<FMyProjectileData>("");
 
 	if (projectileData)
 	{
+		Mesh = projectileData->Mesh;
+
 		ProjectileType = projectileData->ProjectileType;
 
 		Damage = projectileData->Damage;
 		ProjectileSpeed = projectileData->ProjectileSpeed;
+		ProjectileMaxSpeed = projectileData->ProjectileMaxSpeed;
 		ProjectileRange = projectileData->ProjectileRange;
 		ProjectileGravity = projectileData->ProjectileGravity;
 		InitialLaunchAngle = projectileData->InitialLaunchAngle;
 		AreaOfEffectDamage = projectileData->AreaOfEffectDamage;
 		AreaOfEffectRadius = projectileData->AreaOfEffectRadius;
 		DamageOverTime = projectileData->DamageOverTime;
+		DamageOverTimeRate = projectileData->DamageOverTimeRate;
 		DamageOverTimeDuration = projectileData->DamageOverTimeDuration;
 	}
+
+	Mesh = CreateDefaultSubobject<UStaticMeshComponent>(TEXT("Mesh"));
+
+	AreaOfEffectHitbox->SetSphereRadius(AreaOfEffectRadius);
+	AreaOfEffectHitbox->SetActive(false);
+
+	ProjectileMovement->bRotationFollowsVelocity = true;
+	ProjectileMovement->InitialSpeed = ProjectileSpeed;
+	ProjectileMovement->MaxSpeed = ProjectileMaxSpeed;
+	ProjectileMovement->ProjectileGravityScale = ProjectileGravity;
+	ProjectileMovement->Velocity.Z = InitialLaunchAngle;
 }
 
 void AMyProjectileBaseClass::BeginPlay()
 {
 	Super::BeginPlay();
+
+	Player = USteveSingleton::GetSteve()->GetPlayerCharacter();
+
+	if (IsValid(Player))
+	{
+		Steve = Cast<AMyPlayer>(Player);
+	}
+
+	StartingLocation = GetActorLocation();
+
+	ProjectileHitbox->OnComponentBeginOverlap.AddDynamic(this, &AMyProjectileBaseClass::OnHitboxOverlapBegin);
+	AreaOfEffectHitbox->OnComponentBeginOverlap.AddDynamic(this, &AMyProjectileBaseClass::OnAOEHitboxOverlapBegin);
 }
 
 void AMyProjectileBaseClass::Tick(float DeltaTime)
 {
 	Super::Tick(DeltaTime);
-}
\ No newline at end of file
+
+	if (FVector::Dist(StartingLocation, GetActorLocation()) > ProjectileRange)
+	{
+		this->Destroy();
+	}
+}
+
+void AMyProjectileBaseClass::OnHitboxOverlapBegin(UPrimitiveComponent* OverlappedComp, AActor* OtherActor, UPrimitiveComponent* OtherComp, int32 OtherBodyIndex, bool bFromSweep, const FHitResult& SweepResult)
+{
+	if (Enemy == OtherActor)
```

Comment:
if (Enemy == OtherActor) could we write theses the other way?
if (OtherActor == Enemy) makes more sense to me this way..